### PR TITLE
fix: allow broader ActiveSupport compatibility

### PR DESCRIPTION
**Pull request by jtmal-signalfx**
This commit removes additional checks applied to the version of ActiveSupport. It was not a direct dependency, nor was it used directly. It's up to the end user of the gem to resolve dependencies like that.

### CODE REVIEW

File: signalfx.gemspec
Comment (rmfitzpatrick) [inline signalfx.gemspec:42]: Would prefer we not remove assistance, let alone support, in these cases. removing the cap should help?

File: signalfx.gemspec
Comment (rmfitzpatrick) [inline signalfx.gemspec:47]: would be helpful to know why this was pinned in general

File: signalfx.gemspec
Comment (jtmal-signalfx) [reply signalfx.gemspec:42]: I understand the intent, but if the same code is already in protobuf, what is changed by adding it to our gemspec?

Comment (rmfitzpatrick) [reply signalfx.gemspec:42]: As attempted to be explained in the comment, since this condition is evaluated at build time I don't believe it's of assistance for those using 2.2.(0,1).  I'm not sure why the <6 restriction was added and defer to @keitwb https://github.com/signalfx/signalfx-ruby/pull/43/files#diff-565c020e7394b6c19aa8a85842735375b305a476bc048cae1ec0d7c8ce4ebd66R41

My thought would be to revert to what was originally there (no ceiling) and as documented if not using bundler.

File: patch-2.2.1.patch
Comment (rmfitzpatrick) [inline patch-2.2.1.patch]: Is this intentional?

File: signalfx.gemspec
Comment (jtmal-signalfx) [reply signalfx.gemspec:42]: Okay I think I now understand what you meant. I agree there's an issue with installing gems using `gem install ...`, be it from a .gem file or from RubyGems.  I understand it's a valid use case. I have looked at other gems to check how they solve the problem, and I think that the general sentiment is to use Bundler, rather than run `gem install` directly, unless it's to install a specific version of a gem.

As a one example, take a look at:
```bash
root@076de655ca44:~/signalfx-ruby# cat /etc/issue
Ubuntu 16.04.7 LTS \n \l

root@076de655ca44:~/signalfx-ruby# ruby -v
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-linux]
root@076de655ca44:~/signalfx-ruby# gem install activesupport
ERROR:  Error installing activesupport:
        i18n requires Ruby version >= 2.3.0.
```
What I take away form this is that even Rails' core team doesn't care about this use case. My intuition here is that `gem install` doesn't resolves gem versions without checking for supported ruby version, and as a result resolves a version which is too high.

I don't think it's reasonable to expect us to solve these issues for our users.

File: patch-2.2.1.patch
Comment (jtmal-signalfx) [reply patch-2.2.1.patch]: corrected, I was patching ruby to get it to install

File: signalfx.gemspec
Comment (rmfitzpatrick) [reply signalfx.gemspec:42]: The issue had been solved in the same spirit protobuf solves and the actual problem as I see it was the addition of the cap to '<6', which should be removed.  Am I missing something?

Comment (rmfitzpatrick) [reply signalfx.gemspec:42]: For older versions of ruby, they need to build the gem themselves as rubygems doesn't provide assistance.  We are trying to help them and lose nothing in the process afaict.

File: signalfx.gemspec
Comment (rmfitzpatrick) [inline signalfx.gemspec]: I think this will set us up for the issue we're experiencing now and fully reverting would be more favorable:
```ruby
active_support_max_version = "< 5" if Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.2.2")
```

Comment (jtmal-signalfx) [reply signalfx.gemspec]: done

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by rmfitzpatrick** [comment]
@jtmal-signalfx wonder if https://travis-ci.org/github/signalfx/signalfx-ruby/jobs/755360125 is revealing fallout from faye/i18n/active support dependency conflicts.

edit: disregard, my oversight

**Comment by jtmal-signalfx** [comment]
@rmfitzpatrick Thank you, I'm looking into it.

**Pull request merged**
